core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java

```diff
@@ -0,0 +1,285 @@
+            throw new IllegalArgumentException("Negative or zero value for poll interval not allowed.");
+        }
+        Objects.requireNonNull(pollOperation, "The poll operation input parameter cannot be null.");
+
+        this.pollInterval = pollInterval;
+        this.pollOperation = pollOperation;
+        this.pollResponse = new PollResponse<>(PollResponse.OperationStatus.NOT_STARTED, null);
+
+        this.fluxHandle = asyncPollRequestWithDelay()
+            .flux()
+            .repeat()
+            .takeUntil(pollResponse -> pollResponse != null && pollResponse.isDone())
+            .share();
+
+        // auto polling start here
+        this.fluxDisposable = fluxHandle.subscribe();
+        this.autoPollingEnabled = true;
+    }
+
+    /**
+     * Create a Poller object with poll interval, poll operation and cancel operation. The polling starts immediately by default and invoke poll operation.
+     *
+     * @param pollInterval Not null poll interval. It ensure that polling happens only once in given pollInterval.
+     * @param pollOperation   to be called by poller. User should never return {@code null}. The response should have valid {@link com.azure.core.util.polling.PollResponse.OperationStatus}
+     * @param cancelOperation cancel operation
+     * @throws NullPointerException If {@code pollerOptions pollOperation} is {@code null}.
+     */
+    public Poller(Duration pollInterval, Function<PollResponse<T>, Mono<PollResponse<T>>> pollOperation, Consumer<Poller> cancelOperation) {
+        this(pollInterval, pollOperation);
+        this.cancelOperation = cancelOperation;
+    }
+
+    /**
+     * Calls Cancel Operation function if provided.
+     * It will only call cancelOperation if {@link com.azure.core.util.polling.PollResponse.OperationStatus} is IN_PROGRESS otherwise it does nothing.
+     *
+     * @throws UnsupportedOperationException when cancel operation is not provided.
+     */
+    public void cancelOperation() throws UnsupportedOperationException {
+        if (this.cancelOperation == null) {
+            throw new UnsupportedOperationException("Cancel operation is not supported on this service/resource.");
+        }
+
+        // We can not cancel an operation if it was never started
+        // It only make sense to call cancel operation if current status IN_PROGRESS.
+        if (this.pollResponse != null && this.pollResponse.getStatus() != PollResponse.OperationStatus.IN_PROGRESS) {
+            return;
+        }
+        //Time to call cancel
+        this.cancelOperation.accept(this);
+    }
+
+    /**
+     * Enable user to subscribe to {@link Flux} and listen to every {@link PollResponse} asynchronously.
+     * The user will start receiving {@link PollResponse} when client subscribe to this Flux.
+     * The poller could still have its own auto polling in action unless user has turned off
+     * auto polling.
+     *
+     * @return poll response as Flux that can be subscribed.
+     */
```

Comment:
/**
     * This method returns a {@link Flux} that can be subscribed to, enabling a subscriber to receive notification of 
     * every {@link PollResponse}, as it is received.
     *
     * @return A Flux that can be subscribed to to receive poll responses as the long-running operation executes.
     */